packages/react/src/xcss-prop/__tests__/xcss-prop.test.tsx

```diff
@@ -260,4 +260,187 @@ describe('xcss prop', () => {
       />
     ).toBeObject();
   });
+
+  it('should accept an object with specific allowed values in TAllowedProperties', () => {
+    function CSSPropComponent({
+      xcss,
+    }: {
+      xcss: XCSSProp<
+        {
+          color: 'color.text.red' | 'color.text.blue';
+        },
+        never
+      >;
+      // As with standard XCSS, invalid XCSS attributes don't error
+      xcssWrong?: XCSSProp<{ color: 'color.text.red'; invalidCSSAttr: 'invalid' }, never>;
+    }) {
+      return <div className={xcss}>foo</div>;
+    }
+
+    expectTypeOf(
+      <>
+        <CSSPropComponent
+          xcss={{
+            color: 'color.text.red',
+          }}
+        />
+        <CSSPropComponent
+          xcss={{
+            // @ts-expect-error - Type '"red"' is not assignable to type '"color.red" | "color.blue"'.
+            color: 'red',
+            // @ts-expect-error - Type 'string' is not assignable to type 'undefined'.
+            backgroundColor: 'color.background',
+          }}
+        />
+      </>
+    ).toBeObject();
+  });
+
+  it('should enforce required properties when TAllowedProperties set to an object', () => {
+    function CSSPropComponent({
+      xcss,
+    }: {
+      xcss: XCSSProp<
+        { color: 'color.text.red' | 'color.text.blue'; borderColor?: 'color.border' },
+        never
+      >;
+    }) {
+      return <div className={xcss}>foo</div>;
+    }
+
+    expectTypeOf(
+      <>
+        <CSSPropComponent
+          // @ts-expect-error — Property 'color' is missing in type '{}' but required in type {color: 'color.red' | 'color.blue'}.
+          xcss={{}}
+        />
+        <CSSPropComponent
+          xcss={{
+            color: 'color.text.red',
+            // Border color is optional
+          }}
+        />
+      </>
+    ).toBeObject();
+  });
+
+  it('should detect and enforce correct pseudos when TAllowedProperties set to an object', () => {
+    function CSSPropComponent({
+      xcss,
+    }: {
+      xcss: XCSSProp<
+        {
+          color: 'color.text.red' | 'color.text.blue';
+          borderColor?: 'color.border';
+          '&:hover': {
+            // overrides valid color values above
+            color: 'color.text.green';
+            // borderColor can still be set, restricted to values above
+          };
+          '&:focus': {
+            // setting color to `never` should prevent it from being set
+            color: never;
+          };
+          // Pseudos are already optional; optional pseudos don't work as expected
+          '&:active'?: object;
+        },
+        never
+      >;
+      xcssInvalid?: XCSSProp<
+        {
+          color: 'color.text.red' | 'color.text.blue';
+        },
+        // @ts-expect-error - Type 'string' is not assignable to type 'never'.
+        '&:hover' | '&:active'
+      >;
+    }) {
+      return <div className={xcss}>foo</div>;
+    }
+
+    const validStyles = cssMap({
```

Comment:
Problem is here I can't set `&:focus` at all, eg.
```tsx
'&:focus': {
  borderColor?: 'color.border';
}
```

Results in this: 
<img width="897" alt="Screenshot 2024-04-19 at 8 51 15 PM" src="https://github.com/atlassian-labs/compiled/assets/136543114/2cfdd012-e68b-4a26-86c3-ffcbcfdf24c4">

It expects `'&:focus'` to have `{ readonly color: never; }` because `color: 'red'` becomes a `requiredProperties` and passed down into `XCSSPseudo<…>`, so that doesn't quite work.  We almost have to remove `requiredProperties` entirely and just replace it with an object…which, frankly, might even just be better entirely.

---

Maybe it's just better to say you can't even have, in both loose, and strict, a non-object-based `XCSSProp<T>`—but I feel like we tried this and it's a rabbit hole…I could be wrong 😅